ci/update-nixpkgs.py

```diff
@@ -0,0 +1,300 @@
+        except GitCommandError as e:
+            return None
+
+        # Check if there are new commits compared to the last day's integration branch.
+        if f"origin/{last_day_integration_branch}" in nixpkgs_repo.refs:
+            diff_index = nixpkgs_repo.git.diff_index(
+                f"origin/{last_day_integration_branch}"
+            )
+
+            if diff_index == "":
+                info(
+                    "No changes compared to the last day's integration branch. Not creating a new PR."
+                )
+                return None
+
+        nixpkgs_repo.git.push("origin", integration_branch, force=True)
+
+        return NixpkgsRebaseResult(
+            upstream_commit=latest_upstream,
+            fork_commit=nixpkgs_repo.refs[f"origin/{branch_to_rebase}"].commit,
+            fork_before_rebase=current_state,
+            fork_after_rebase=nixpkgs_repo.head.commit,
+        )
+
+    info("Nothing to do.")
+
+
+def update_fc_nixos(
+    target_branch: str,
+    integration_branch: str,
+    previous_hex_sha: str,
+    new_hex_sha: str,
+):
+    info(f"Update fc-nixos.")
+    repo = Repo(Path.cwd())
+    if not any(integration_branch == head.name for head in repo.heads):
+        tracking_branch = repo.create_head(
+            integration_branch, f"origin/{target_branch}"
+        )
+        tracking_branch.checkout()
+    else:
+        repo.git.checkout(integration_branch)
+
+    check_output(
+        [
+            "nix",
+            "flake",
+            "lock",
+            "--override-input",
+            "nixpkgs",
+            f"github:{NIXPKGS_REPO}/{new_hex_sha}",
+        ]
+    )
+    check_output(["nix", "run", ".#buildVersionsJson"]).decode("utf-8")
+    check_output(["nix", "run", ".#buildPackageVersionsJson"]).decode("utf-8")
+
+    changelog_path = (
+        Path(CHANGELOG_DIR)
+        / f"{datetime.datetime.now().strftime('%Y%m%d_%H%M%S')}_nixpkgs-auto-update-{target_branch}.md"
+    )
+    changelog_path.write_text(
+        f"""
+### NixOS XX.XX platform
+
+- Update nixpkgs from {previous_hex_sha} to {new_hex_sha}
+"""
+    )
+
+    repo.git.add(
+        [
+            "flake.lock",
+            VERSIONS_PATH,
+            PACKAGE_VERSIONS_PATH,
+            str(changelog_path),
+        ]
+    )
+    repo.git.commit(message=f"Auto update nixpkgs to {new_hex_sha}")
+    repo.git.push("origin", integration_branch, force=True)
+
+
+def create_fc_nixos_pr(
+    target_branch: str,
+    integration_branch: str,
+    github_access_token: str,
+    now: str,
+):
+    info(f"Create PR in fc-nixos.")
+    gh = Github(auth=Auth.Token(github_access_token))
+    fc_nixos_repo = gh.get_repo(FC_NIXOS_REPO)
+    fc_nixos_repo.create_pull(
+        base=target_branch,
+        head=integration_branch,
+        title=f"Auto update nixpkgs {now}",
+        body=f"""\
+View nixpkgs update branch: [{integration_branch}](https://github.com/{NIXPKGS_REPO}/tree/{integration_branch})
```

Comment:
More more guidance how to deal with this PR, general workflow and edge cases (merge/rebase conflicts)